### PR TITLE
Add regfish.de

### PR DESCRIPTION
**Pull request by gertfriend**
New features

    Added support for regfish.de DynamicDNS

### CODE REVIEW

**Review by satrapes — COMMENTED** [review]
It is very hard for me to review because the documentation of the DynDNS is in German and an automatic translate tool failed to help.
However, I have one general comment. It seems like the pull request only addresses 1 single use case (ipv4) and it doesn't handle all different use cases (i.e. ipv6, ttl, forcehost etc) and doesn't address potential error inputs.

**Comment by gertfriend** [comment]
I hope this helps:
[Manual Update.pdf](https://github.com/ddclient/ddclient/files/9593769/Manual.Update.pdf)

**Comment by satrapes** [comment]
Yes it does, thanks I will try to do a more thorough review at some point.

File: ddclient.in [hunk 1]
```diff
@@ -7111,7 +7111,7 @@ sub nic_regfishde_update {
 
    # No response, declare as failed
    if (!defined($reply) || !$reply) {
-     failed("regfishde updating %s: Could not connect to %s.", $h, $config{$h}{'server
+     failed("regfis.hde updating %s: Could not connect to %s.", $h, $config{$h}{'server
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply ddclient.in:7114]:
```suggestion
     failed("regfish.de updating %s: Could not connect to %s.", $h, $config{$h}{'server
```

File: ddclient.in [hunk 2]
```diff
@@ -7055,6 +7068,70 @@ sub nic_keysystems_update {
...
+ debug("\nnic_regfishdev4_update -------------------");
+
+ ## update each configured host
+ ## should improve to update in one pass
+ for my $h (@_) {
+   my $ip = delete $config{$h}{'wantip'};
+   info("regfish.de setting IP address to %s for %s", $ip, $h);
+
+   my $url = "https://$config{$h}{'server'}/?fqdn=$h&ipv4=$ip&forcehost=1&token=$config
+{$h}{'password'}";
```
Comment (SuperSandro2000) [inline ddclient.in:7107]:
```suggestion
   my $url = "https://$config{$h}{'server'}/?fqdn=$h&ipv4=$ip&forcehost=1&token=$config{$h}{'password'}";
```

File: ddclient.in [hunk 3]
```diff
@@ -7055,6 +7068,70 @@ sub nic_keysystems_update {
...
+######################################################################
+## nic_regfishdev4_examples
+######################################################################
+sub nic_regfishdev4_examples {
+    return <<EoEXAMPLE;
+o 'regfishde'
+The 'regfishde' protocol is used by the non-free
+dynamic DNS service offered by www.regfish.de.
+???Check https://www.regfish.de for configuration variables applicable
+to the 'regfishde' protocol are:
```
Comment (SuperSandro2000) [inline ddclient.in:7080]:
```suggestion
The 'regfishde' protocol is used by the non-free dynamic DNS service offered by www.regfish.de.
Check https://www.regfish.de for configuration variables applicable to the 'regfishde' protocol:
```

**Comment by gertfriend** [comment]
Finally had time to try it and after minor changes it works so please merge it.

File: ddclient.in [hunk 4]
```diff
@@ -7055,6 +7068,73 @@ sub nic_keysystems_update {
...
+ for my $h (@_) {
+   my $ip = delete $config{$h}{'wantip'};
+   info("regfish.de setting IP address to %s for %s", $ip, $h);
+
+   my $url;
+   $url .= "https://$config{$h}{'server'}/?";
+   $url .= "fqdn=$h";
+   $url .= "&ipv4=$ip";
+   $url .= "&forcehost=1";
+   $url .= "&token=$config{$h}{'password'}";
```
Comment (SuperSandro2000) [inline ddclient.in:7111]:
```suggestion
   my $url = "https://$config{$h}{'server'}/?fqdn=$h&ipv4=$ip&forcehost=1&token=$config{$h}{'password'}";
```

File: ddclient.in [hunk 5]
```diff
@@ -7055,6 +7068,73 @@ sub nic_keysystems_update {
...
+   # No response, declare as failed
+   if (!defined($reply) || !$reply) {
+     failed("regfish.de updating %s: Could not connect to %s.", $h, $config{$h}{'server'});
+     last;
+   }
+   last if !header_ok($h, $reply);
+
+   if ($reply =~ /success/)
+   {
+       $config{$h}{'ip'}   = $ip;
```
Comment (SuperSandro2000) [inline ddclient.in:7125]:
```suggestion
       $config{$h}{'ip'} = $ip;
```

**Comment by gertfriend** [comment]
Do i have to do anything else so that the code is merged?

**Comment by SuperSandro2000** [comment]
> Do i have to do anything else so that the code is merged?

I think after those 3 last small changes I am going to merge it. Also the merge conflict needs to be resolved.

**Pull request closed without merging**